docs/ui/storage/fragments/web/s3-image.md

```diff
@@ -0,0 +1,198 @@
+import { AppComponent } from './app.component';
+
+import { AmplifyUIAngularModule } from '@aws-amplify/ui-angular';
+import Amplify from 'aws-amplify';
+import awsconfig from './aws-exports';
+
+Amplify.configure(awsconfig);
+
+@NgModule({
+  declarations: [AppComponent],
+  imports: [AmplifyUIAngularModule, BrowserModule],
+  providers: [],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+_app.component.html_
+
+```html
+<amplify-s3-image img-key="example.png" />
+```
+</docs-filter>
+
+<docs-filter framework="ionic">
+
+_app.module.ts_
+
+```js
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { AppComponent } from './app.component';
+
+import { AmplifyUIAngularModule } from '@aws-amplify/ui-angular';
+import Amplify from 'aws-amplify';
+import awsconfig from './aws-exports';
+
+Amplify.configure(awsconfig);
+
+@NgModule({
+  declarations: [AppComponent],
+  imports: [AmplifyUIAngularModule, BrowserModule],
+  providers: [],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+_app.component.html_
+
+```html
+<amplify-s3-image img-key="example.png" />
+```
+</docs-filter>
+
+<docs-filter framework="vue">
+
+_main.js_
+
+```js
+import Vue from 'vue';
+import App from './App.vue';
+import '@aws-amplify/ui-vue';
+import Amplify from 'aws-amplify';
+import awsconfig from './aws-exports';
+
+Amplify.configure(awsconfig);
+
+new Vue({
+  render: h => h(App),
+}).$mount('#app');
+```
+
+_App.vue_
+
+```html
+<template>
+  <amplify-s3-image img-key="example.png" />
+</template>
+```
+</docs-filter>
+
+
+<ui-component-props tag="amplify-s3-image" use-table-headers></ui-component-props>
+
+
+## Use Cases
+
+### Getting files from protected access level
+
+Protected files from S3 bucket can be accessed by anyone other than the owner by using `identityId` of the owner.
+
+<docs-filter framework="react">
+
+```jsx
+const App = () => (
+ <AmplifyS3Image level="protected" imgKey="example.png" identityId="us-east-1:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX"/>
+);
+```
+</docs-filter>
+
+<docs-filter framework="angular">
+
+_app.component.html_
+
+```html
+ <amplify-s3-image level="protected" img-key="example.png" identity-id="us-east-1:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX"/>
+```
+</docs-filter>
+
+<docs-filter framework="ionic">
+
+_app.component.html_
+
+```html
+ <amplify-s3-image level="protected" img-key="example.png" identity-id="us-east-1:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX"/>
+```
+</docs-filter>
+
+<docs-filter framework="vue">
+
+_App.vue_
+
+```html
+<template>
+   <amplify-s3-image level="protected" img-key="example.png" identity-id="us-east-1:XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX"/>
+</template>
+```
+</docs-filter>
+
+## Customization
+
+### Custom CSS Properties
+
+`amplify-s3-image` provides the following CSS properties to modify the style at component level.
+
+| Name           | Description                      | Default value                    |
```

Comment:
Q for @jordanranz, would this fit into the table component we have for [current documentation](https://docs.amplify.aws/ui/auth/authenticator/q/framework/react#props-amplify-authenticator)?

Reply:
This is how we have done in auth docs. i just followed it. Not sure if there is a comp for this.

Reply:
yup yup, I dont know if there was a comp we introduced for it so can follow up with Jordan on it

Reply:
The component level CSS API is new to the docs. There currently is not component that will *tableify* this but it would be a huge improvement to add this. To build this we would need to add comments to the scss in the component lib, then ingest that in a docs stencil component similar to the `ui-component-props` component